source/keymanweb.js

```diff
@@ -3542,16 +3572,102 @@
     {
       if(document.attachEvent)
         document.attachEvent('onfocusin', keymanweb._IEFocusIn);
-      else if(document.addEventListener)
-        document.addEventListener('DOMNodeInserted', keymanweb._DOMNodeInserted, true);
     }
    
     // Restore and reload the currently selected keyboard 
     keymanweb.restoreCurrentKeyboard(); 
 
+      /* Setup of handlers for dynamically-added and (eventually) dynamically-removed elements.
+       * Reference: https://developer.mozilla.org/en/docs/Web/API/MutationObserver
+       * 
+       * We place it here so that it loads after most of the other UI loads, reducing the MutationObserver's overhead.
+       */
+
+      keymanweb.observationTarget = document.querySelector('body');
+      keymanweb.mutationObserver = new MutationObserver(function(mutations) 
+        {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(i=0; i < mutations.length; i++)
+          {
+            mutation = mutations[i];
+            
+            for(j = 0; j < mutation.addedNodes.length; j++)
+            {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = ""
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) 
+              {
+                var arr = addedNode.getElementsByTagName('input');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+
+                arr = addedNode.getElementsByTagName('textarea');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+
+                arr = addedNode.getElementsByTagName('iframe');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+              }
+
+              if(lcTagName == 'input' || lcTagName == 'textarea' || lcTagName == 'iframe')
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(addedNode);
+              }
+
+              for(k = 0; k < childAdditions.length; k++)
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(childAdditions[k]);
+              }
+            }
+
+            // There also exists a 'mutation.removedNodes' array.  We'll need to address that for issue #63.
+
+            // After all mutations have been handled, we need to recompile our .sortedInputs array.
+            if(dirtyFlag) keymanweb.listInputs();
+          }
+        });
+
+      keymanweb.observationConfig = { childList: true, subtree: true };
```

Comment:
just inline this, or declare it as a local var, because it isn't referenced elsewhere?